### PR TITLE
feature: replace order

### PR DESCRIPTION
**Pull request by lispc**

### CODE REVIEW

File: src/state/global_state.rs [hunk 1]
```diff
@@ -505,23 +531,23 @@ impl GlobalState {
         encoded_tx[tx_detail_idx::AY1] = account.ay;
         encoded_tx[tx_detail_idx::NONCE1] = account.nonce;
         encoded_tx[tx_detail_idx::BALANCE1] = proof.leaf;
-        encoded_tx[tx_detail_idx::ORDER1_AMOUNT_SELL] = tx.previous_amount_sell;
-        encoded_tx[tx_detail_idx::ORDER1_AMOUNT_BUY] = tx.previous_amount_buy;
-        encoded_tx[tx_detail_idx::ORDER1_FILLED_SELL] = tx.previous_filled_sell;
-        encoded_tx[tx_detail_idx::ORDER1_FILLED_BUY] = tx.previous_filled_buy;
+        encoded_tx[tx_detail_idx::ORDER1_AMOUNT_SELL] = old_order.tokensell;
```
Comment (HAOYUatHZ) [inline src/state/global_state.rs:534]: `old_order.total_sell`?

File: src/state/global_state.rs [hunk 2]
```diff
@@ -505,23 +531,23 @@ impl GlobalState {
...
+        encoded_tx[tx_detail_idx::ORDER1_AMOUNT_BUY] = old_order.tokenbuy;
```
Comment (HAOYUatHZ) [inline src/state/global_state.rs:535]: `old_order.total_buy`?

File: src/state/global_state.rs (hunk 1 above)
Comment (lispc) [reply src/state/global_state.rs:534]: Yes…

File: src/state/common.rs [hunk 3]
```diff
@@ -27,13 +27,23 @@ impl Order {
...
+
+impl Order {
     pub fn hash(&self) -> Fr {
         let mut data = Fr::zero();
         data.add_assign(&self.status);
         data.add_assign(&shl(&self.tokenbuy, 32));
         data.add_assign(&shl(&self.tokensell, 64));
         hash(&[data, self.filled_sell, self.filled_buy, self.total_sell, self.total_buy])
     }
+    pub fn is_filled(&self) -> bool {
```
Comment (HAOYUatHZ) [inline src/state/common.rs:40]: This logic isn't quite right.

Comment (HAOYUatHZ) [reply src/state/common.rs:40]: https://github.com/Loopring/protocols/blob/5e49e7b10e80238fbf71822e2e1bf110af686872/packages/loopring_v3/circuit/Gadgets/MatchingGadgets.h#L224-L257

**Pull request merged**